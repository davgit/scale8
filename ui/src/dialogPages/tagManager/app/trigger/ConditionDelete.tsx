import { FC } from 'react';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { DialogMutationFunction } from '../../../abstractions/DialogDirectMutation';
import DeleteConditionRuleQuery from '../../../../gql/mutations/DeleteConditionRuleQuery';
import { DialogDeleteComment } from '../../../abstractions/DialogDeleteComment';

const ConditionDelete: FC<DialogPageProps> = (props: DialogPageProps) => {
    return (
        <DialogDeleteComment
            text={`Delete Condition ${props.name}?`}
            executeMutationCallback={async (
                mutationFunction: DialogMutationFunction,
                id: string,
                contextId: string,
                ids: string[],
                comments: string,
            ) => {
                await mutationFunction({
                    variables: {
                        conditionRuleDeleteInput: {
                            condition_rule_id: id,
                            ...(comments === '' ? {} : { comments }),
                        },
                    },
                });
            }}
            mutation={DeleteConditionRuleQuery}
            {...props}
        />
    );
};

export default ConditionDelete;
