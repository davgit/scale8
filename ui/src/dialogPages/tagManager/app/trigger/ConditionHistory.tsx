import { FC } from 'react';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { DialogEntityHistory } from '../../../abstractions/DialogEntityHistory';

const ConditionHistory: FC<DialogPageProps> = (props: DialogPageProps) => {
    return <DialogEntityHistory infoKeyBase="conditions" {...props} />;
};

export default ConditionHistory;
