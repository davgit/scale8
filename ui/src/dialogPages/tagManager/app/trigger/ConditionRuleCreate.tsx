import { FC } from 'react';
import { useMutation, useQuery } from '@apollo/client';
import CreateConditionRuleQuery from '../../../../gql/mutations/CreateConditionRuleQuery';
import ConditionRuleForm from '../../../../components/organisms/Forms/ConditionRuleForm';
import FetchAvailableDataContainersQuery from '../../../../gql/queries/FetchAvailableDataContainersQuery';
import { FetchAvailableDataContainers } from '../../../../gql/generated/FetchAvailableDataContainers';
import { ConditionMode } from '../../../../gql/generated/globalTypes';
import {
    buildConditionRuleCreateInput,
    ConditionRuleFormProps,
    ConditionRuleValues,
    useConditionRuleForm,
} from '../../../../hooks/form/useConditionRuleForm';
import { SelectValueWithSub } from '../../../../hooks/form/useFormValidation';
import { DataContainer } from '../../../../types/DataMapsTypes';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { usePageDialogControls } from '../../../../hooks/dialog/usePageDialogControls';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { QueryLoaderAndError } from '../../../../abstractions/QueryLoaderAndError';
import { getDataContainersIcon } from '../../../../utils/TypeIconsUtils';
import { useConfigState } from '../../../../context/AppContext';
import { logError } from '../../../../utils/logUtils';

type ConditionRuleCreateProps = DialogPageProps & {
    submitText: string;
    title: string;
    infoKeyBase: string;
    conditionMode: ConditionMode;
};

type ConditionRuleCreateAfterLoadProps = ConditionRuleCreateProps & {
    availableDataContainers: DataContainer[];
    consentPurposes: { id: number; name: string }[];
    consentVendors: { id: number; name: string }[];
    dataContainersSelectValues: SelectValueWithSub[];
};

const ConditionRuleCreateAfterLoad: FC<ConditionRuleCreateAfterLoadProps> = (
    props: ConditionRuleCreateAfterLoadProps,
) => {
    const [createConditionRule, { loading, data, error: gqlError }] =
        useMutation(CreateConditionRuleQuery);

    const formInitialState = {
        name: '',
        dataContainerId: '',
        match: '',
        matchId: '',
        matchKey: '',
        matchCondition: '',
        dataMapValue: '',
        comments: '',
    };

    const submitForm = async (conditionRuleValues: ConditionRuleValues) => {
        const conditionRuleCreateInput = buildConditionRuleCreateInput(
            props.id,
            props.conditionMode,
            conditionRuleValues,
        );

        try {
            await createConditionRule({
                variables: { conditionRuleCreateInput },
            });
        } catch (error) {
            logError(error);
        }
    };

    const conditionRuleFormValues = useConditionRuleForm(
        formInitialState,
        submitForm,
        props.availableDataContainers,
    );

    const successfullySubmitted = data?.createConditionRule.id !== undefined;

    usePageDialogControls(
        JSON.stringify(formInitialState) === JSON.stringify(conditionRuleFormValues.values),
        successfullySubmitted,
        props.setPageHasChanges,
        props.handleDialogClose,
        props.pageRefresh,
    );

    if (loading || successfullySubmitted) {
        return <div />;
    }

    const formProps: ConditionRuleFormProps = {
        ...conditionRuleFormValues,
        gqlError,
        submitText: props.submitText,
        title: props.title,
        formInfoProps: buildStandardFormInfo(props.infoKeyBase, 'Create'),
        handleDialogClose: props.handleDialogClose,
        dataContainers: props.dataContainersSelectValues,
        consentPurposes: props.consentPurposes,
        consentVendors: props.consentVendors,
    };

    return <ConditionRuleForm {...formProps} />;
};

const ConditionRuleCreate: FC<ConditionRuleCreateProps> = (props: ConditionRuleCreateProps) => {
    const { consentPurposes, consentVendors } = useConfigState();

    return QueryLoaderAndError<FetchAvailableDataContainers>(
        false,
        useQuery<FetchAvailableDataContainers>(FetchAvailableDataContainersQuery, {
            variables: { triggerId: props.id },
        }),
        (data: FetchAvailableDataContainers) => {
            return (
                <ConditionRuleCreateAfterLoad
                    availableDataContainers={data.getTrigger.revision.app_platform_revisions.reduce(
                        (accumulator: DataContainer[], currentValue) => {
                            return [
                                ...accumulator,
                                ...currentValue.platform_revision.platform_data_containers,
                            ] as DataContainer[];
                        },
                        [],
                    )}
                    dataContainersSelectValues={data.getTrigger.revision.app_platform_revisions.map(
                        (_) => ({
                            key: _.platform_revision.platform.id,
                            text: _.platform_revision.platform.name,
                            sub: _.platform_revision.platform_data_containers.map((c) => {
                                const Icon = getDataContainersIcon(c.icon);

                                return {
                                    key: c.id,
                                    text: c.name,
                                    iconType: c.icon,
                                    icon: <Icon />,
                                    description: c.description,
                                };
                            }) as SelectValueWithSub[],
                        }),
                    )}
                    consentPurposes={consentPurposes}
                    consentVendors={consentVendors}
                    {...props}
                />
            );
        },
    );
};

export { ConditionRuleCreate };
