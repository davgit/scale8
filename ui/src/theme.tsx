import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
