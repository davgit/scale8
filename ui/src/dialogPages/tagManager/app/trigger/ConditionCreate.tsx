import { FC } from 'react';
import { ConditionRuleCreate } from './ConditionRuleCreate';
import { ConditionMode } from '../../../../gql/generated/globalTypes';
import { DialogPageProps } from '../../../../types/DialogTypes';

const ConditionCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    return (
        <>
            <ConditionRuleCreate
                {...props}
                submitText={'Add Condition'}
                title={'Add Condition'}
                conditionMode={ConditionMode.CONDITION}
                infoKeyBase={'conditions'}
            />
        </>
    );
};

export default ConditionCreate;
