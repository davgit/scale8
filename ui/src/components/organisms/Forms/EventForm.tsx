import { useTheme } from '@mui/material/styles';
import { FC, useEffect, useState } from 'react';
import { Checkbox, FormControlLabel } from '@mui/material';
import ControlledFilteredSelects from '../../atoms/ControlledInputs/ControlledFilteredSelects';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import { SelectValueWithSub } from '../../../hooks/form/useFormValidation';
import { controlledSelectValuesFindByInnerKey } from '../../../utils/ControlledSelectUtils';
import { MappedPlatformValuesForm } from '../../molecules/MappedPlatformValues/MappedPlatformValuesForm';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import ControlledTextAreaInput from '../../atoms/ControlledInputs/ControlledTextAreaInput';
import ControlledIntegerInput from '../../atoms/ControlledInputs/ControlledIntegerInput';
import {
    EventFormProps,
    EventValues,
} from '../../../dialogPages/tagManager/app/trigger/EventCreate';
import { buildEventName } from '../../../dialogPages/tagManager/app/trigger/EventUpdate';

const getDescription = (values: EventValues, availableEvents: SelectValueWithSub[]): string => {
    if (!values.eventId) {
        return '';
    }

    const selectedEvent = controlledSelectValuesFindByInnerKey(availableEvents, values.eventId);

    return selectedEvent?.description ?? '';
};

const EventForm: FC<EventFormProps> = (props: EventFormProps) => {
    const theme = useTheme();
    const notAvailable = props.availableEvents.length < 1;

    const [useCustom, setUseCustom] = useState(!!props.useCustom);
    const [generateName, setGenerateName] = useState(
        props.generateName === undefined ? true : props.generateName,
    );
    const [useEventAsName, setUseEventAsName] = useState(
        props.useEventAsName === undefined ? true : props.useEventAsName,
    );

    const { values, availableEvents, handleChange } = props;

    useEffect(() => {
        if (useCustom) {
            const currentName = values.browserEvent;

            if (useEventAsName && values.name !== currentName) {
                handleChange('name', currentName);
            }
        } else {
            const currentName = buildEventName(values.eventId, availableEvents);

            if (generateName && values.name !== currentName) {
                handleChange('name', currentName);
            }
        }
    }, [values, availableEvents, handleChange, generateName, useEventAsName, useCustom]);

    return (
        <DrawerFormLayout {...props}>
            <FormControlLabel
                control={
                    <Checkbox
                        disabled={!!props.update}
                        name="useCustom"
                        checked={useCustom}
                        onChange={(event) => {
                            if (event.target.checked) {
                                handleChange('eventId', '');
                                setGenerateName(true);
                                setUseCustom(true);
                            } else {
                                handleChange('eventId', '');
                                setUseEventAsName(true);
                                setUseCustom(false);
                            }
                        }}
                        color="primary"
                    />
                }
                label="Use Custom Browser Event"
            />
            {useCustom ? (
                <>
                    <ControlledTextInput
                        disabled={!!props.update}
                        name="browserEvent"
                        label="Custom Event"
                        formProps={props}
                        className="DrawerFormField"
                        required
                    />
                </>
            ) : notAvailable ? (
                <small>There are no more events available for this rule.</small>
            ) : (
                <>
                    <ControlledFilteredSelects
                        disabled={!!props.update}
                        className="DrawerFormField"
                        initialFilterValue={props.initialPlatformId}
                        label="Event"
                        name="eventId"
                        values={availableEvents}
                        formProps={props}
                        required
                        filterLabel="Platform"
                        missingSubMessage="There are no more events available from this platform."
                        hideNoSub
                    />
                    <small
                        style={{
                            width: '100%',
                            margin: theme.spacing(0, 0, 2),
                        }}
                    >
                        {getDescription(values, availableEvents)}
                    </small>
                    {props.values.mappedPlatformValues !== undefined &&
                        props.values.eventId !== '' && (
                            <MappedPlatformValuesForm
                                mappedPlatformValues={props.values.mappedPlatformValues}
                                parentLocators={[]}
                                {...props}
                            />
                        )}
                </>
            )}

            <FormControlLabel
                control={
                    <Checkbox
                        name="useClearState"
                        checked={values.clearState !== -1}
                        onChange={(event) => {
                            if (event.target.checked) {
                                handleChange('clearState', 1000);
                            } else {
                                handleChange('clearState', -1);
                            }
                        }}
                        color="primary"
                    />
                }
                label="Use timer to reset event state"
            />
            {values.clearState !== -1 && (
                <ControlledIntegerInput
                    name="clearState"
                    label="Reset event state after (milliseconds)"
                    formProps={props}
                    className="DrawerFormField"
                    required
                />
            )}

            {useCustom ? (
                <>
                    <FormControlLabel
                        style={{ marginBottom: theme.spacing(3) }}
                        control={
                            <Checkbox
                                name="useEventAsName"
                                checked={useEventAsName}
                                onChange={(event) => {
                                    setUseEventAsName(event.target.checked);
                                }}
                                color="primary"
                            />
                        }
                        label="Use Custom Event as Name"
                    />
                    {!useEventAsName && (
                        <ControlledTextInput
                            name="name"
                            label="Name"
                            formProps={props}
                            className="DrawerFormField"
                            required
                        />
                    )}
                </>
            ) : notAvailable ? null : (
                <>
                    {props.values.mappedPlatformValues !== undefined &&
                        props.values.eventId !== '' && (
                            <>
                                <FormControlLabel
                                    style={{ marginBottom: theme.spacing(3) }}
                                    control={
                                        <Checkbox
                                            name="generateName"
                                            checked={generateName}
                                            onChange={(event) => {
                                                setGenerateName(event.target.checked);
                                            }}
                                            color="primary"
                                        />
                                    }
                                    label="Generate Name"
                                />
                                {!generateName && (
                                    <ControlledTextInput
                                        name="name"
                                        label="Name"
                                        formProps={props}
                                        className="DrawerFormField"
                                        required
                                    />
                                )}
                            </>
                        )}
                </>
            )}
            <ControlledTextAreaInput
                name="comments"
                label="Comments"
                formProps={props}
                className="DrawerFormField"
            />
        </DrawerFormLayout>
    );
};

export default EventForm;
