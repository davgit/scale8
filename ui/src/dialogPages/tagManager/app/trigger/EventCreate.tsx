import { FC } from 'react';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import CreateEventQuery from '../../../../gql/mutations/CreateEventQuery';
import { DataMapInput } from '../../../../gql/generated/globalTypes';
import EventForm from '../../../../components/organisms/Forms/EventForm';
import FetchAvailableEventsQuery from '../../../../gql/queries/FetchAvailableEventsQuery';
import { mappedPlatformValuesToDataMapInput } from '../../../../utils/MappedPlatformValuesUtils';
import { FormWithMappedPlatformValuesResult } from '../../../../hooks/form/useFormWithMappedPlatformValues';
import { SelectValueWithSub } from '../../../../hooks/form/useFormValidation';
import nameValidator from '../../../../utils/validators/nameValidator';
import { ValuesWithMappedPlatformData } from '../../../../types/MappedPlatformValuesTypes';
import { CreateEventResult } from '../../../../gql/generated/CreateEventResult';
import { PlatformDataMap } from '../../../../types/DataMapsTypes';
import { PlatformEvent } from '../../../../types/TagRulesTypes';
import { DialogPageProps } from '../../../../types/DialogTypes';
import { buildStandardFormInfo } from '../../../../utils/InfoLabelsUtils';
import { FetchAvailableEvents } from '../../../../gql/generated/FetchAvailableEvents';
import { FormBaseProps } from '../../../../types/props/forms/CommonFormProps';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../../abstractions/DialogPreloadForm';
import { getEventIcon } from '../../../../utils/TypeIconsUtils';
import { useConfigState } from '../../../../context/AppContext';

export type EventBaseValues = {
    eventId: string;
    browserEvent: string;
    comments: string;
    name: string;
    clearState: number;
};

export type EventValues = ValuesWithMappedPlatformData<EventBaseValues>;

export type EventFormProps = FormWithMappedPlatformValuesResult<EventValues> &
    FormBaseProps & {
        update?: boolean;
        useCustom?: boolean;
        useEventAsName?: boolean;
        generateName?: boolean;
        initialPlatformId?: string;
        availableEvents: SelectValueWithSub[];
        consentPurposes: { id: number; name: string }[];
        consentVendors: { id: number; name: string }[];
    };

const EventCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const { consentPurposes, consentVendors } = useConfigState();

    const eventCreateProps: DialogPreloadFormProps<
        FetchAvailableEvents,
        EventValues,
        EventFormProps,
        CreateEventResult
    > = {
        loadQuery: useQuery<FetchAvailableEvents>(FetchAvailableEventsQuery, {
            variables: { triggerId: props.id },
        }),
        buildInitialStatePreload: () => ({
            eventId: '',
            browserEvent: '',
            name: '',
            comments: '',
            clearState: -1,
        }),
        saveQuery: useMutation(CreateEventQuery),
        mapSaveData: (eventValues: EventValues) => {
            const data_maps: DataMapInput[] = mappedPlatformValuesToDataMapInput(
                eventValues.mappedPlatformValues ?? [],
            );

            return {
                eventCreateInput: {
                    trigger_id: props.id,
                    platform_event_id: eventValues.eventId === '' ? undefined : eventValues.eventId,
                    browser_event:
                        eventValues.browserEvent === '' ? undefined : eventValues.browserEvent,
                    name: eventValues.name,
                    clear_state_ms: eventValues.clearState,
                    data_maps,
                    ...(eventValues.comments === ''
                        ? {}
                        : {
                              comments: eventValues.comments,
                          }),
                },
            };
        },
        buildFormProps: (
            formLoadedData: FetchAvailableEvents,
            formValidationValues: FormWithMappedPlatformValuesResult<EventValues>,
            gqlError?: ApolloError,
        ): EventFormProps => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Add Event',
            title: 'Add Event',
            formInfoProps: buildStandardFormInfo('events', 'Create'),
            handleDialogClose: props.handleDialogClose,
            availableEvents: formLoadedData.getTrigger.revision.app_platform_revisions.map((_) => ({
                key: _.platform_revision.platform.id,
                text: _.platform_revision.platform.name,
                sub: _.platform_revision.platform_events.map((e) => {
                    const Icon = getEventIcon(e.icon);

                    return {
                        key: e.id,
                        text: e.name,
                        iconType: e.icon,
                        icon: <Icon />,
                        description: e.description,
                    };
                }) as SelectValueWithSub[],
            })),
            consentPurposes,
            consentVendors,
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createEvent.id !== undefined,
        pageComponent: EventForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Event name too short',
            },
        ],
        mappedPlatformValuesProps: {
            buildAvailableModelsWithPlatformDataMaps: (formLoadedData: FetchAvailableEvents) => {
                const availableEvents =
                    formLoadedData.getTrigger.revision.app_platform_revisions.reduce(
                        (accumulator, currentValue) => {
                            return [
                                ...accumulator,
                                ...(currentValue.platform_revision
                                    .platform_events as PlatformEvent[]),
                            ];
                        },
                        [] as PlatformEvent[],
                    );
                return availableEvents.map((_) => ({
                    id: _.id,
                    platformDataMaps: _.platform_data_maps as PlatformDataMap[],
                }));
            },
            idValueForModelWithPlatformDataMaps: 'eventId',
        },
        ...props,
    };

    return (
        <DialogPreloadForm<FetchAvailableEvents, EventValues, EventFormProps, CreateEventResult>
            {...eventCreateProps}
        />
    );
};

export default EventCreate;
