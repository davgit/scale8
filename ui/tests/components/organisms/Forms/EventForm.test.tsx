import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import {
    platformDataFormResultProps,
    useMockFormProps,
} from '../../../componentsMocks/useMockFormProps';
import EventForm from '../../../../src/components/organisms/Forms/EventForm';
import { EventValues } from '../../../../src/dialogPages/tagManager/app/trigger/EventCreate';

const MockForm: FC = () => {
    const formProps = useMockFormProps<EventValues>({
        eventId: '',
        browserEvent: '',
        comments: '',
        name: '',
        clearState: 0,
    });

    return (
        <EventForm
            {...formProps}
            {...platformDataFormResultProps}
            availableEvents={[{ key: 'p1', text: 'p1' }]}
            consentPurposes={[]}
            consentVendors={[]}
        />
    );
};

describe('EventForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
