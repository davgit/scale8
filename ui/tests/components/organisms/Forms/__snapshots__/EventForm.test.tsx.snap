// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="useCustom"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Use Custom Browser Event
            </span>
          </label>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Event
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk Mui-error css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-eventId"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-eventId"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="eventId"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium Mui-required css-1sbojr7-MuiFormHelperText-root"
            >
              Please select a value
            </p>
          </div>
          <small
            class="MuiBox-root css-nbwlf2"
          />
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                name="useClearState"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
            >
              Use timer to reset event state
            </span>
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Reset event state after (milliseconds)
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                name="clearState"
                required=""
                type="text"
                value="0"
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Comments
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-1yxqjki-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <textarea
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                name="comments"
                style="height: 0px; overflow: hidden;"
              />
              <textarea
                aria-hidden="true"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                readonly=""
                style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                tabindex="-1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-dhh0px"
                >
                  <span>
                    Comments
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
